esphome/components/deep_sleep/__init__.py

```diff
@@ -167,6 +194,18 @@ def validate_config(config):
     }
 )
 
+WakeupPinItem = deep_sleep_ns.struct("WakeupPinItem")
+WAKEUP_SINGLEPIN_SCHEMA = cv.All(pins.internal_gpio_input_pin_schema)
```

Comment:
`cv.All` is not needed when there is only one validation
```suggestion
WAKEUP_SINGLEPIN_SCHEMA = pins.internal_gpio_input_pin_schema
```